src/doc/rustc/src/platform-support/riscv32e-unknown-none-elf.md

```diff
@@ -0,0 +1,30 @@
+# `riscv32{e,em,emc}-unknown-none-elf`
+
+**Tier: 3**
+
+Bare-metal target for RISC-V CPUs with the RV32E, RV32EM and RV32EMC ISAs.
+
+## Target maintainers
+
+* Henri Lunnikivi, <henri.lunnikivi@gmail.com>, [@hegza](https://github.com/hegza)
+
+## Requirements
+
+The target is cross-compiled, and uses static linking. No external toolchain is
+required and the default `rust-lld` linker works, but you must specify a linker
+script.
+
+## Building the target
+
+This target is included in Rust and can be installed via `rustup`.
+
+## Testing
+
+This is a cross-compiled `no-std` target, which must be run either in a
+simulator or by programming them onto suitable hardware. It is not possible to
+run the Rust test-suite on this target.
+
+## Cross-compilation toolchains and C code
+
+This target supports C code. If interlinking with C or C++, you may need to use
+`riscv64-unknown-elf-gcc` as a linker instead of `rust-lld`.
```

Comment:
"may"? hm. unclear why this linker, specifically? if it's a cross-target linker, shouldn't it be `riscv32-unknown-elf-gcc`?

Reply:
Source is strictly copy-paste from the [riscv32-unknown-none-elf target specification](src/doc/rustc/src/platform-support/riscv32-unknown-none-elf.md) but I have indeed seen this work in practice. I'm usually able to make my C compilers `--with-multilib` which means that either `riscv32-unknown-elf-gcc` or `riscv64-unknown-elf-gcc` will generally work.

Since both of them work, and this is a 32-bit target, perhaps it could be more correct to use `riscv32-unknown-elf-gcc` here. However, my intuition tells me that _in practice_ the `riscv64-unknown-elf-gcc` would work more often than `riscv32`.

Hmm... I'll change it to `riscv32-unknown-elf-gcc` which I feel is more correct.

Reply:
My layman's experience is that generally we have one (`riscv32-*-gcc`) on half of our platforms and the other one (`riscv64-*-gcc`) the other half. Both of them work just fine but you need to use the right name for the tool or the compile will fail, so it just becomes a point of contention which one we should use in build tools and CI 🙂

Reply:
Huh. Weird.